test/cast_test.exs

```diff
@@ -6,6 +6,69 @@ defmodule OpenApiSpec.CastTest do
 
   def cast(ctx), do: Cast.cast(ctx)
 
+  describe "SubScheme with array" do
+    defmodule User do
+      defstruct [:email, :permissions]
+    end
+
+    defmodule Permission do
+      defstruct [:name, :some_other_field]
+    end
+
+    defmodule UsersSchemas do
+      alias OpenApiSpex.Schema
+
+      defmodule PermissionSchema do
+        @moduledoc false
+        require OpenApiSpex
+
+        OpenApiSpex.schema(%{
+          title: "Permission",
+          type: :object,
+          properties: %{
+            name: %Schema{type: :string}
+          },
+          required: [:name, :some_other_field]
+        })
+      end
+
+      defmodule UserSchema do
+        @moduledoc false
+        require OpenApiSpex
+
+        OpenApiSpex.schema(%{
+          title: "User",
+          type: :object,
+          properties: %{
+            email: %Schema{type: :string},
+            permissions: %Schema{type: :array, items: PermissionSchema}
```

Comment:
`items` should be an explicit [%Reference{}](https://hexdocs.pm/open_api_spex/OpenApiSpex.Reference.html), or use `OpenApiSpex.resolve_schema_modules` to convert it to one automatically.